docs/DeveloperGuide.md

```diff
@@ -4,26 +4,120 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+The bulk of the app's work is done by the following three components:
+- `UI`: The UI of the App.
+- `Parser`: Formats the user's input
+- `Command`: Command's logic and execution
+- `Storage`: Storage of data of the App
+
+![Architecture Diagram](./images/ArchitectureDiagram.png "Architecture Diagram")
+
+### UI component
+
+![UI Sequence Diagram](./images/cs2113-ui-sequence.png "UI Sequence Diagram")
+![UI Class Diagram](./images/ui-class-diagram.png "UI Class Diagram")
+
+The `UI` consists of a `Scanner` and an `OutputStream` object. Together, these objects abstract the functionalities of
+obtaining user input and providing feedback (output printed in terminal UI). The `UI` component provides a simple
+interface for other components to interact with the user.
+
+The `UI` component:
+- provides a method to obtain user input.
+- provide methods to print output in tabular format
+
+### Parser component
+
+The `Parser` functionality is to take in a user input, parse it and return the relevant `Command` object based on 
+the input.
+
+How the `Parser` works:
+1. When the user input any string, it will be passed to a newly constructed `Parser` object.
+2. The `parse` function in the `Parser` will be called to extract the command word, description and arguments of the 
+command if any.
+3. These parsed details will then be passed to the relevant `Command` object based on the command word.
+4. Afterward, the `Command` object will be passed back to `Main` for execution.
+
+Note: The `Parser` will not do any validation of arguments or description of the command. Those will be handled by the 
+respective `Command` object.
+
+![Parser Sequence Diagram](./images/ParserSequence.png "Parser Sequence Diagram")
+
+
+### Command component
+
+The Command component consists of the individual command objects (listed in table below) and an abstract
+class `Command`. The `Command` component is responsible for executing the commands after it has been parsed by `Parser`. \
+All error handling is handled here and any errors/output would be passed to the `UI` component for printing and
+formatting of the output.
+
+| Command                  | Purpose                                    |
```

Comment:
Clear summary for command components! Perhaps some sample inputs could be shown to demonstrate usage?